onnxruntime/core/optimizer/selectors_actions/actions.h

```diff
@@ -158,6 +158,8 @@ struct ReplaceWithNew : public Action {
   // specifies how the inputs and outputs for the replaced nodes are moved to the new node
   virtual std::vector<NodeAndMoveInfo> ValueMoves(const RuntimeState&) const = 0;
 
+  virtual Status ProcessNewNode(Graph&, const NodesToOptimize&, Node&) const { return Status::OK(); }
```

Comment:
could you add some comments describing when this is called and what it should do? e.g., it should do any additional processing required on the replacement node when we directly applying or replaying the action (not saving the action).